packages/components/src/checkbox-control/test/index.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import BaseCheckboxControl from '..';
+import type { CheckboxControlProps } from '../types';
+
+const getInput = () => screen.getByTestId( 'checkbox' );
```

Comment:
Instead of applying a test id to the markup, let's write the tests from the point of view of our users — and therefore let's use roles whenever possible:

```suggestion
const getInput = () => screen.getByRole( 'checkbox' ) as HTMLInputElement;
```